common/src/main/java/io/homeassistant/companion/android/common/sensors/DisplaySensorManager.kt

```diff
@@ -98,4 +149,104 @@ class DisplaySensorManager : SensorManager {
             mapOf()
         )
     }
+
+    private fun updateScreenOrientation(context: Context) {
+        if (!isEnabled(context, screenOrientation)) {
+            return
+        }
+
+        @Suppress("DEPRECATION")
+        val orientation = when (context.resources.configuration.orientation) {
+            Configuration.ORIENTATION_PORTRAIT -> "portrait"
+            Configuration.ORIENTATION_LANDSCAPE -> "landscape"
+            Configuration.ORIENTATION_SQUARE -> "square"
+            Configuration.ORIENTATION_UNDEFINED -> STATE_UNKNOWN
+            else -> STATE_UNKNOWN
+        }
+
+        val icon = when (orientation) {
+            "portrait" -> "mdi:phone-rotate-portrait"
+            "landscape" -> "mdi:phone-rotate-landscape"
+            "square" -> "mdi:crop-square"
+            else -> screenOrientation.statelessIcon
+        }
+
+        onSensorUpdated(
+            context,
+            screenOrientation,
+            orientation,
+            icon,
+            mapOf()
+        )
+    }
+
+    private fun updateScreenRotation(context: Context) {
+        if (!isEnabled(context, screenRotation)) {
+            return
+        }
+
+        val wm = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
+
+        @Suppress("DEPRECATION")
+        val display = when (wm.defaultDisplay.rotation) {

```

Comment:
This is deprecated which makes me wonder: what is the default display on a foldable? When folded open is it landscape or portrait and if it is landscape would you expect to still see portrait because the cover display hasn't rotated?

Reply:
Latest commit will no longer use this deprecated method, instead the state of the sensor will always be of the default display. There will be an attribute for rotation angle of additional displays on those devices.